src/main/java/com/ibm/stocator/fs/ObjectStoreFileSystem.java

```diff
@@ -218,10 +218,24 @@ public boolean rename(Path src, Path dst) throws IOException {
       return true;
     }
     LOG.debug("Checking if source exists {}", src);
+
     if (exists(src)) {
-      LOG.debug("Source {} exists", src);
+      boolean result = storageClient.rename(hostNameScheme, src.toString(), dst.toString());
+
```

Comment:
@gilv Can you explain what you mean? If src does not exist, then we return false without doing anything.